### PR TITLE
[FIX] web: prevent crash when serializing action button's context

### PR DESCRIPTION
**Pull request by sdegueldre**
Previously, when evaluating a button's context from its attributes, we
would evaluate it with the record's data as context, this resulted in
x2many fields being evaluated as a datapoint (in particular, as
StaticList). When we then went to call doAction with that context, the
action service tries to serialize that context with JSON.stringify,
which crashes as the datapoint contains circular references. This is a
mistake, as contexts and domain that are to be evaluated from a
string/attribute should be evaluated using the record's evalContext, not
its data. This commit fixes that.

### CODE REVIEW

File: addons/web/static/tests/views/form/form_view_tests.js [hunk 1]
```diff
@@ -5940,6 +5940,47 @@ QUnit.module("Views", (hooks) => {
...
+    QUnit.test("clicking on a stat button with x2many in context", async function (assert) {
+        assert.expect(1);
+        serverData.models.partner.records[1].timmy = [12];
+
+        const actionService = {
+            start() {
+                return {
+                    doActionButton(args) {
+                        // button context should have been evaluated
+                        // the x2many should be in the for of x2m commands
```
Comment (Polymorphe57) [inline addons/web/static/tests/views/form/form_view_tests.js:5952]: in the form

File: addons/web/static/src/views/view_button/view_button_hook.js [hunk 2]
```diff
@@ -56,7 +56,7 @@ export function useViewButtons(model, ref, options = {}) {
                 let buttonContext = {};
                 if (clickParams.context) {
                     if (typeof clickParams.context === "string") {
```
Comment (Polymorphe57) [inline addons/web/static/src/views/view_button/view_button_hook.js:58]: datapoint (in particular, as
StaticList)
--> a StaticList

Comment (Polymorphe57) [reply addons/web/static/src/views/view_button/view_button_hook.js:58]: the datapoint contains --> the static list...

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**